cel/cel_test.go

```diff
@@ -1190,38 +1194,296 @@ func TestCustomInterpreterDecorator(t *testing.T) {
+		},
+		{
+			name:    "minus",
+			program: `1 - 1`,
+			wanted:  oneCost,
+		},
+		{
+			name:    "/",
+			program: `1 / 1`,
+			wanted:  oneCost,
+		},
+		{
+			name:    "/",
+			program: `1 * 1`,
+			wanted:  oneCost,
+		},
+		{
+			name:    "%",
+			program: `1 % 1`,
+			wanted:  oneCost,
+		},
+		{
+			name:    "ternary",
+			program: `true ? 1 : 2`,
+			wanted:  oneCost,
+		},
+		{
+			name:    "string size",
+			program: `size("123")`,
+			wanted:  oneCost,
+		},
+		{
+			name:    "bytes to string conversion",
+			decls:   []*exprpb.Decl{decls.NewVar("input", decls.Bytes)},
+			hints:   map[string]int64{"input": 500},
+			program: `string(input)`,
+			wanted:  checker.CostEstimate{Min: 1, Max: 51},
+		},
+		{
+			name:    "string to bytes conversion",
+			decls:   []*exprpb.Decl{decls.NewVar("input", decls.String)},
+			hints:   map[string]int64{"input": 500},
+			program: `bytes(input)`,
+			wanted:  checker.CostEstimate{Min: 1, Max: 51},
+		},
+		{
+			name:    "int to string conversion",
+			program: `string(1)`,
+			wanted:  checker.CostEstimate{Min: 1, Max: 1},
+		},
+		{
+			name:    "contains",
+			program: `input.contains(arg1)`,
+			decls: []*exprpb.Decl{
+				decls.NewVar("input", decls.String),
+				decls.NewVar("arg1", decls.String),
+			},
+			hints:  map[string]int64{"input": 500, "arg1": 500},
+			wanted: checker.CostEstimate{Min: 2, Max: 2502},
+		},
+		{
+			name:    "matches",
+			program: `input.matches('\\d+a\\d+b')`,
+			decls: []*exprpb.Decl{
+				decls.NewVar("input", decls.String),
+			},
+			hints:  map[string]int64{"input": 500},
+			wanted: checker.CostEstimate{Min: 1, Max: 101},
+		},
+		{
+			name:    "startsWith",
+			program: `input.startsWith(arg1)`,
+			decls: []*exprpb.Decl{
+				decls.NewVar("input", decls.String),
+				decls.NewVar("arg1", decls.String),
+			},
+			hints:  map[string]int64{"arg1": 500},
+			wanted: checker.CostEstimate{Min: 2, Max: 52},
+		},
+		{
+			name:    "endsWith",
+			program: `input.endsWith(arg1)`,
+			decls: []*exprpb.Decl{
+				decls.NewVar("input", decls.String),
+				decls.NewVar("arg1", decls.String),
+			},
+			hints:  map[string]int64{"arg1": 500},
+			wanted: checker.CostEstimate{Min: 2, Max: 52},
+		},
+		{
+			name:    "size receiver",
+			program: `input.size()`,
+			decls: []*exprpb.Decl{
+				decls.NewVar("input", decls.String),
+			},
+			wanted: checker.CostEstimate{Min: 2, Max: 2},
+		},
+		{
+			name:    "size",
+			program: `size(input)`,
+			decls: []*exprpb.Decl{
+				decls.NewVar("input", decls.String),
+			},
+			wanted: checker.CostEstimate{Min: 2, Max: 2},
+		},
 	}
-	min, max = EstimateCost(prg)
-	if min != wantedCost[0] || max != wantedCost[1] {
-		log.Fatalf("Got cost interval [%v, %v], wanted %v",
-			min, max, wantedCost)
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			if tc.hints == nil {
+				tc.hints = map[string]int64{}
+			}
+			descriptor := new(proto3pb.TestAllTypes).ProtoReflect().Descriptor()
```

Comment:
I don't think you're actually using the `TestAllTypes` proto in this test, so you can remove this line and the `DeclareContextProto()` option below.

Reply:
I'm using `TestAllTypes` to represent message types in the tests.

Reply:
Ah, right. Though I think the `DeclareContextProto` isn't doing what you expect. It should be publishing a bunch of declarations for all of the type-level fields in the proto.

Reply:
Oh, I get it. I've switched this to use `Types(...)`